### PR TITLE
Fix adding histograms

### PR DESCRIPTION
**Pull request by bryates**
This PR partially fixes the disagreement with the old `HistEFT` during the datacard stage. There still is some disagreement in a handful of bins, sometimes in the overflow. I'm still trying to debug that. @btovar does this change make sense, or is there a more elegant solution?

### CODE REVIEW

**Review by btovar — COMMENTED** [review]
It worries me that this changes the meaning of the [] operator. If I do

a["x"] = 2
a["x"] = 4

I'd expect the value of a["x"] to be 4, not 6. Probably this kind of fix should be done in the __iadd__ method.

**Comment by bryates** [comment]
> It worries me that this changes the meaning of the [] operator. If I do
> 
> a["x"] = 2 a["x"] = 4
> 
> I'd expect the value of a["x"] to be 4, not 6. Probably this kind of fix should be done in the **iadd** method.

That's a great point. I'll look into modifying `__iadd__` instead.

File: topcoffea/modules/sparseHist.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def __setitem__(self, key, value):
             h = list(filtered.values())[0]
 
         try:
-            h[nocats] = value
+            h[nocats] = value.values()
```
[the remark this answers is not in the document]
Comment (btovar) [reply topcoffea/modules/sparseHist.py:290]: Looks good!

**Comment by kmohrman** [comment]
If @btovar is happy with this change, then I would be happy to merge it.

**Pull request merged**